### ISSUE TITLE
Usage of Clock.systemDefaultZone() should not trigger [JavaTimeDefaultTimeZone]

### ISSUE THREAD

**Issue by ChristianCiach** [google/error-prone#1312]
### Description of the problem / feature request:
Using the static method `java.time.Clock.systemDefaultZone()` triggers the warning 
> [JavaTimeDefaultTimeZone] Clock.systemDefaultZone() is not allowed because it silently uses the system default time-zone. (..)

I think it's a bit funny that calling a method named "systemDefaultZone()" causes a warning containing the phrase "it silently uses the system default time-zone".

I could probably workaround the warning by using `Clock.system(ZoneId.systemDefault())`, but I would argue that I shouldn't have to do this. I think `Clock.systemDefaultZone()` is just as descriptive as the workaround, if not even more so.

### What version of Error Prone are you using?
2.3.3

**Comment by sumitbhagwani** [comment]
The point is to use a fixed ZoneId rather than `ZoneId.systemDefault()`
Documentation says "You must choose an explicit ZoneId."

perhaps we should warn on usage of `ZoneId.systemDefault()` too

**Comment by tbroyer** [comment]
I disagree, the point is about not using APIs that **silently** use the default timezones. It shouldn't flag **explicit** uses of the default timezones.

**Comment by tbroyer** [comment]
That said, in this case this is explicitly checked for, so I doubt this change request will be taken into account:
https://github.com/google/error-prone/blob/369005c0b3d29b09eeeebc70693425ec986d1954/core/src/test/java/com/google/errorprone/bugpatterns/time/JavaTimeDefaultTimeZoneTest.java#L32
